src/modules/headset/gamepad.c

```diff
@@ -0,0 +1,189 @@
+#include "headset/headset.h"
+#include "core/os.h"
+#include "core/util.h"
+#include "core/maf.h"
+#include <stdlib.h>
+#define OS_GLFW_H
+#include "core/os_glfw.h"
+
+#define GAMEPAD_COUNT (DEVICE_GAMEPAD_LAST - DEVICE_GAMEPAD_FIRST + 1)
+
+typedef struct {
+  bool present;
+  int jid;
+} GamepadState;
+
+static struct {
+  bool inited;
+  GamepadState gamepad[GAMEPAD_COUNT];
+  int gamepadsPresent;
+} state;
+
+static GamepadState *getGamepad(Device device) {
+  if (device >= DEVICE_GAMEPAD_FIRST && device <= DEVICE_GAMEPAD_LAST) {
+    return &state.gamepad[device - DEVICE_GAMEPAD_FIRST];
+  }
+  return NULL;
+}
+
+void discoverGamepads() {
+  for(int jid = 0; jid < GLFW_JOYSTICK_LAST; jid++) {
+    if (glfwJoystickPresent(jid)) {
```

Comment:
I use `glfwJoystickPresent()`, but maybe we should use `glfwJoystickIsGamepad()`. It is not clear to me exactly which devices say yes to "present" but not to "isGamepad".

Reply:
GLFW docs say if you're only interested in gamepads, you can use isGamepad as a replacement for isPresent